src/controllers/dynakube/activegate/internal/capability/reconciler_test.go

```diff
@@ -1,517 +1,360 @@
+		verifyReconciler(t, r)
+
+		service := &corev1.Service{}
+		err := r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, service)
+		require.Error(t, err)
+		assert.NotNil(t, service)
+
+		err = r.createOrUpdateService()
+		require.NoError(t, err)
+
+		service = &corev1.Service{}
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, service)
+		require.NoError(t, err)
+		assert.NotNil(t, service)
+	})
+	t.Run(`Update works for ports`, func(t *testing.T) {
+		clt := createClient()
+		instance := createInstance()
+		mockStatefulSetReconciler := getMockReconciler()
+		mockCustompropertiesReconciler := getMockReconciler()
+
+		r := NewReconciler(clt, capability.NewMultiCapability(instance), instance, mockStatefulSetReconciler, mockCustompropertiesReconciler)
+		verifyReconciler(t, r)
+
+		err := r.createOrUpdateService()
+		require.NoError(t, err)
+
+		service := &corev1.Service{}
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, service)
+		require.NoError(t, err)
+		assert.NotNil(t, service)
+
+		tmpService := &corev1.Service{}
+
+		service.Spec.Ports = []corev1.ServicePort{}
+
+		err = r.createOrUpdateService()
+		require.NoError(t, err)
+
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, tmpService)
+		require.NoError(t, err)
+		assert.NotNil(t, tmpService)
+
+		require.NotEqual(t, tmpService, service)
+	})
+	t.Run(`update works for labels`, func(t *testing.T) {
+		clt := createClient()
+		instance := createInstance()
+		mockStatefulSetReconciler := getMockReconciler()
+		mockCustompropertiesReconciler := getMockReconciler()
+
+		r := NewReconciler(clt, capability.NewMultiCapability(instance), instance, mockStatefulSetReconciler, mockCustompropertiesReconciler)
+		verifyReconciler(t, r)
+
+		err := r.createOrUpdateService()
+		require.NoError(t, err)
+
+		service := &corev1.Service{}
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, service)
+		require.NoError(t, err)
+		assert.NotNil(t, service)
+
+		tmpService := &corev1.Service{}
+
+		service.Labels = map[string]string{}
+
+		err = r.createOrUpdateService()
+		require.NoError(t, err)
+
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, tmpService)
+		require.NoError(t, err)
+		assert.NotNil(t, service)
+
+		require.NotEqual(t, tmpService, service)
+	})
+}
+
+func TestPortsAreOutdated(t *testing.T) {
+	t.Run(`portsAreOutdated works`, func(t *testing.T) {
```

Comment:
method name is redundant, this should describe what the test is doing, e.g. "ports are detected as outdated"